FirebaseRemoteConfig/Sources/RCNConfigRealtime.m

```diff
@@ -353,20 +365,24 @@ - (void)setUpHttpSession {
 
 #pragma mark - Retry Helpers
 
+- (BOOL)canMakeConnection {
+  BOOL noRunningConnection =
+      self->_dataTask == nil || self->_dataTask.state != NSURLSessionTaskStateRunning;
+  BOOL canMakeConnection = noRunningConnection && [self->_listeners count] > 0 &&
+                           !self->_isInBackground && !self->_isRealtimeDisabled;
+  return canMakeConnection;
+}
+
 // Retry mechanism for HTTP connections
 - (void)retryHTTPConnection {
   __weak RCNConfigRealtime *weakSelf = self;
   dispatch_async(_realtimeLockQueue, ^{
     __strong RCNConfigRealtime *strongSelf = weakSelf;
-    if (strongSelf->_isInBackground) {
+    if (!strongSelf || strongSelf->_isInBackground) {
```

Comment:
redundant, but ok for clarity. I'm not sure if there is style guidance on this.

Reply:
I did this under the impression that `strongSelf->_isInBackground` would crash if `strongSelf == nil`.

Reply:
In ObjC a message passed to nil returns nil

Reply:
ohh, thanks. 🤦 I thought that the `->` syntax was more unsafe and would crash when the receiver was nil. I just tested and see that it doesn't. I'll leave this for clarity then.